proxy/Main.cc

```diff
@@ -476,6 +476,18 @@ check_config_directories(void)
   }
 }
 
+namespace {
+  // Values with names that have this prefix are exempted from being considered "unrecognized".
+  const char PLUGIN_CONFIG_RECORD_PREFIX[] = "proxy.config.plugin.";
```

Comment:
This is a big assumption. I have internal plugins that use configuration values that are not in this namespace.